src/sql/workbench/common/editor/query/queryEditorInput.ts

```diff
@@ -346,6 +321,28 @@ export abstract class QueryEditorInput extends EditorInput implements IConnectab
 			}
 		}
 		this._onDidChangeLabel.fire();
+
+		this.contextualizeEditorForCopilot();
+	}
+
+	private contextualizeEditorForCopilot(): void {
+		// Don't need to take any actions if contextualization is not enabled and can return
+		if (!this.configurationService.getValue<IQueryEditorConfiguration>('queryEditor').githubCopilotContextualizationEnabled) {
+			return;
+		}
+
+		this.serverContextualizationService.getServerContextualization(this.uri)
+			.then(async getServerContextualizationResult => {
```

Comment:
Using `then` instead of `await` because `await` will wait for all of these async operations to complete. Using `then` allows the following async operations to split off from the original call and allow them to complete at a later time that the original scope isn't able to predict.

Reply:
I'm a little confused - doesn't everything in the `.then()` happen after `this.serverContextualizationService.getServerContextualization(this.uri)` is resolved? For awaiting multiple promises in parallel, shouldn't `await Promise.all()` be used?

Reply:
@lewis-sanchez It'd probably be a bit simpler to make this function async and use awaits, but then just have the caller not await on the function to complete (since it's essentially just a background task)

Having nested then's is a bit more annoying to follow and doesn't really provide any actual value here.